### DISCUSSION TITLE
Why using asyncio on BatchHandler._postprocess_batch()?

### DISCUSSION THREAD

**Discussion by aiorga-sherpas** [michaelfeil/infinity#129] [General]
Greetings Michael,

we noticed the use of the ThreadPoolExecutor to run the preprocessing_batch and the core_buch but to execute the postprocess Asycio is used. Is there any particular reason for this? 
There is also the following comment: # TODO: the ugly asyncio.sleep() could add to 3-8ms of latency worst case ...
Moving all asynchronous processes to threadpool futures might solve this problem?

**Comment by michaelfeil** [comment]
Great catch! Short answer: I am currently putting 'asyncio.Futures' in the queue - asyncio.futures are not threadsafe.

Asyncio.sleep - I'll verify to which degree they are helpful / if the speedup is still valid across all platforms supported nowadays. The sleep is primarily because the other threads are starting to execute at the same time & not block the model thread (Python GIL)
